csrc/host_ir/executor.h

```diff
@@ -86,7 +86,8 @@ class HostIrExecutor final : public OptInDispatch {
   // Cache Fusions, FusionExecutors
   std::unordered_map<HostUnit*, FusionExecutor> fe_;
   std::unordered_map<HostUnit*, FusionExecutorCache> fec_;
-  std::unordered_map<Stream*, c10::cuda::CUDAStream> streams_;
+  using StreamKey = std::variant<int64_t, Stream*>;
```

Comment:
I'm personally not a big fan of std::variant because it often ends up mixing two things in one data structure. In this particular case, it's probably easier to understand and debug to keep the original `Stream*=>CUDAStream` mapping and add another `std::unordered_map<int64_t,Stream*>` or `std::vector<Stream*>` (if indices are compact) for only the named streams.

Reply:
I understand, however I personally prefer using it here. I applied your suggestion in [the last commit](https://github.com/NVIDIA/Fuser/pull/2621/commits/1d7be069fcffbc789f1dd2411ae21b7bf4441056), but I would personally revert as it makes the code more cumbersome and duplicated.

Also, even though I get your point, we are already relying a lot on `std::variant`, through dynamic type lib and `PolymorphicValue`

Let me know whether to keep the last commit.

Reply:
Yes, PolymorphicValue uses std::variant and it comes with a cost. For example, I found it hard to gdb the value and had to deal with dynamic types in a statically-typed language. 

Unlike dynamic_type, I failed to see large enough benefits to justify the cost. So I'm inclined to avoid std::variant.

Reply:
The cost is the diff of this commit: https://github.com/NVIDIA/Fuser/pull/2621/commits/3f84fa6fafa8d074931dbe99f6e819772bff462b

The code is more cumbersome, and duplicated in many points. Also, IMO, harder to debug.

Conversely, I don't see the benefit of not using `std::variant` in the present case. Even when using gdb, having to deal with one key and one hashtable is easier IMO, even if the key is of variant type (but with one single print we can see what type and value it is, without having to guess ti what hashtable the key belongs)